### PR TITLE
Add table oci devops project

### PR DESCRIPTION
**Pull request by lucasjellema**
This PR is a fix for issue #487 

I have a problem with the way the tests are implemented: JSON documents are stringified and then compared. For some reason, it happens that the logically same JSON documents are stringified differently and when compared are judged different making the test fail  - incorrectly. The test should probably order the JSON document by key before doing the stringify (or so ChatGPT says)

# Integration test logs
<details>
  <summary>Logs</summary>
gitpod /workspace/steampipe-plugin-oci/oci-test (add-table-oci-devops-project) $ node ./tint.js oci_devops_project
Test names: [ 'tests/oci_devops_project' ]
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_devops_project []

PRETEST: tests/oci_devops_project

TEST: tests/oci_devops_project
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # oci_devops_project.named_test_resource will be created
  + resource "oci_devops_project" "named_test_resource" {
      + compartment_id    = "ocid1.tenancy.oc1..aaaaaaaag7c7slwmlvsodyym662ixlsonnihko2igwpjwwe2egmlf3gg6okq"
      + defined_tags      = (known after apply)
      + description       = (known after apply)
      + freeform_tags     = (known after apply)
      + id                = (known after apply)
      + lifecycle_details = (known after apply)
      + name              = "steampipetest6079"
      + namespace         = (known after apply)
      + state             = (known after apply)
      + system_tags       = (known after apply)
      + time_created      = (known after apply)
      + time_updated      = (known after apply)

      + notification_config {
          + topic_id = (known after apply)
        }
    }

  # oci_ons_notification_topic.named_test_ons_topic_resource will be created
  + resource "oci_ons_notification_topic" "named_test_ons_topic_resource" {
      + api_endpoint   = (known after apply)
      + compartment_id = "ocid1.tenancy.oc1..aaaaaaaag7c7slwmlvsodyym662ixlsonnihko2igwpjwwe2egmlf3gg6okq"
      + defined_tags   = (known after apply)
      + description    = "Terraform testing resource"
      + etag           = (known after apply)
      + freeform_tags  = {
          + "Name" = "steampipetest6079"
        }
      + id             = (known after apply)
      + name           = "steampipetest6079"
      + short_topic_id = (known after apply)
      + state          = (known after apply)
      + time_created   = (known after apply)
      + topic_id       = (known after apply)
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + notification_topic_id = (known after apply)
  + reporting_region      = "us-ashburn-1"
  + resource_id           = (known after apply)
  + resource_name         = "steampipetest6079"
  + tenancy_ocid          = "ocid1.tenancy.oc1..aaaaaaaag7c7slwmlvsodyym662ixlsonnihko2igwpjwwe2egmlf3gg6okq"
oci_ons_notification_topic.named_test_ons_topic_resource: Creating...
oci_ons_notification_topic.named_test_ons_topic_resource: Creation complete after 1s [id=ocid1.onstopic.oc1.iad.aaaaaaaaapffk6xtldm6iimwvlszivufzysfut67spo3s6x36xdxif6xog3a]
oci_devops_project.named_test_resource: Creating...
oci_devops_project.named_test_resource: Creation complete after 2s [id=ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caatsrxwcos2lfi3oxoxg3kfj3zu7kishzukvewp3yupl3a]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

notification_topic_id = "ocid1.onstopic.oc1.iad.aaaaaaaaapffk6xtldm6iimwvlszivufzysfut67spo3s6x36xdxif6xog3a"
reporting_region = "us-ashburn-1"
resource_id = "ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caatsrxwcos2lfi3oxoxg3kfj3zu7kishzukvewp3yupl3a"
resource_name = "steampipetest6079"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaag7c7slwmlvsodyym662ixlsonnihko2igwpjwwe2egmlf3gg6okq"

Running SQL query: test-get-query.sql

[
  {
    "id": "ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caatsrxwcos2lfi3oxoxg3kfj3zu7kishzukvewp3yupl3a",
    "name": "steampipetest6079"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "id": "ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caatsrxwcos2lfi3oxoxg3kfj3zu7kishzukvewp3yupl3a",
    "name": "steampipetest6079"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaag7c7slwmlvsodyym662ixlsonnihko2igwpjwwe2egmlf3gg6okq",
    "title": "steampipetest6079"
    "title": "steampipetest6079",
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaag7c7slwmlvsodyym662ixlsonnihko2igwpjwwe2egmlf3gg6okq"
  }
]

✘ FAILED

TEARDOWN: tests/oci_devops_project
</details>

# Example query results
<details>
  <summary>Results</summary>

steampipe query "select   
>   name,
>   id, 
>   lifecycle_state, 
>   notification_topic_id, 
>   namespace, 
>   lifecycle_details, 
>   description, 
>   time_created, 
>   time_updated, 
>   tags 
> from   
>   oci_devops_project;
> "
```+----------------------------------+------------------------------------------------------------------------------------------+-----------------+-------------------------------------------------------------------------------------+--------------+-------------------+------------------------------------------------+----------------------+----------------------+--------------------------------------------------------------------------------------------------------------------+
| name                             | id                                                                                       | lifecycle_state | notification_topic_id                                                               | namespace    | lifecycle_details | description                                    | time_created         | time_updated         | tags                                                                                                               |
+----------------------------------+------------------------------------------------------------------------------------------+-----------------+-------------------------------------------------------------------------------------+--------------+-------------------+------------------------------------------------+----------------------+----------------------+--------------------------------------------------------------------------------------------------------------------+
| HelloWorldFn_6Pwp_devops_project | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caaufehl36qak6ojt3d5zmpcw7wbevupbev76h4fkj6tdwq | ACTIVE          | ocid1.onstopic.oc1.iad.aaaaaaaa2sc3hpsvfkv2trgkk6nj5heazi277qbdnsmomur5f4ha4vh6u7mq | idtwlqf2hanz | <null>            | HelloWorldFn_6Pwp_devops_project               | 2021-10-03T11:19:54Z | 2021-10-03T11:47:48Z | {"CreatedBy":"oracleidentitycloudservice/jellema@amis.nl","CreatedOn":"2021-10-03T11:19:54.607Z","release":"1.11"} |
| go-on-oci                        | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caa2b3unelei22uswncl5qxllcfdgft2nyg7ypabs5qpbka | ACTIVE          | ocid1.onstopic.oc1.iad.aaaaaaaalkegqtyhk3sqmjrrn2cy4x2bsmqxlrd4mpw7dx4hd6cmahtvt6cq | idtwlqf2hanz | <null>            | Resources for the Way to Go on OCI application | 2022-04-04T05:55:14Z | 2022-05-23T06:52:01Z | {"CreatedBy":"oracleidentitycloudservice/jellema@amis.nl","CreatedOn":"2022-04-04T05:55:13.898Z"}                  |
| devopsproject_zApq               | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caafqen7vittabcfzqm6ayyqwedisa2ymvblfdva2rgznpq | ACTIVE          | ocid1.onstopic.oc1.iad.aaaaaaaag4rsikhsiewgp63fcimxgabmr6iuk2nhkrkaks6bujw6gd2wj64a | idtwlqf2hanz | <null>            | DevOps Project for Instance Group deployment   | 2021-10-03T12:07:13Z | 2021-10-13T17:04:59Z | {"CreatedBy":"oracleidentitycloudservice/jellema@amis.nl","CreatedOn":"2021-10-03T12:07:13.254Z","release":"1.1"}  |
| cloudnative-2021_devops_project  | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caa34xp275akik3pw3ykergwbk7tkfhox576q44aqambpaa | ACTIVE          | ocid1.onstopic.oc1.iad.aaaaaaaak3z7owbl5hgrsoq3izf6yyqusyjmb64cffj2736t3okteffkv7ma | idtwlqf2hanz | <null>            | cloudnative-2021_devops_project                | 2021-11-25T12:26:24Z | 2021-12-02T06:33:23Z | {"CreatedBy":"oracleidentitycloudservice/jellema@amis.nl","CreatedOn":"2021-11-25T12:26:24.482Z"}                  |
+----------------------------------+------------------------------------------------------------------------------------------+-----------------+-------------------------------------------------------------------------------------+--------------+-------------------+------------------------------------------------+----------------------+----------------------+--------------------------------------------------------------------------------------------------------------------+```

gitpod /workspace/steampipe-plugin-oci (add-table-oci-devops-project) $ steampipe query "select                                    
  project.name,
  project.id, 
  project.notification_topic_id, 
  topic.name notification_topic_name,
  project.namespace, 
  project.description
from   
  oci_devops_project project
  left join
  oci_ons_notification_topic topic
  on project.notification_topic_id = topic.topic_id
"
```+----------------------------------+------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+-------------------------+--------------+------------------------------------------------+
| name                             | id                                                                                       | notification_topic_id                                                               | notification_topic_name | namespace    | description                                    |
+----------------------------------+------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+-------------------------+--------------+------------------------------------------------+
| devopsproject_zApq               | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caafqen7vittabcfzqm6ayyqwedisa2ymvblfdva2rgznpq | ocid1.onstopic.oc1.iad.aaaaaaaag4rsikhsiewgp63fcimxgabmr6iuk2nhkrkaks6bujw6gd2wj64a | zApq_devopstopic        | idtwlqf2hanz | DevOps Project for Instance Group deployment   |
| HelloWorldFn_6Pwp_devops_project | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caaufehl36qak6ojt3d5zmpcw7wbevupbev76h4fkj6tdwq | ocid1.onstopic.oc1.iad.aaaaaaaa2sc3hpsvfkv2trgkk6nj5heazi277qbdnsmomur5f4ha4vh6u7mq | HelloWorldFn_6Pwp_topic | idtwlqf2hanz | HelloWorldFn_6Pwp_devops_project               |
| cloudnative-2021_devops_project  | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caa34xp275akik3pw3ykergwbk7tkfhox576q44aqambpaa | ocid1.onstopic.oc1.iad.aaaaaaaak3z7owbl5hgrsoq3izf6yyqusyjmb64cffj2736t3okteffkv7ma | cloudnative-2021_topic  | idtwlqf2hanz | cloudnative-2021_devops_project                |
| go-on-oci                        | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caa2b3unelei22uswncl5qxllcfdgft2nyg7ypabs5qpbka | ocid1.onstopic.oc1.iad.aaaaaaaalkegqtyhk3sqmjrrn2cy4x2bsmqxlrd4mpw7dx4hd6cmahtvt6cq | devops-topic            | idtwlqf2hanz | Resources for the Way to Go on OCI application |
+----------------------------------+------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+-------------------------+--------------+------------------------------------------------+```
  
gitpod /workspace/steampipe-plugin-oci (add-table-oci-devops-project) $ steampipe query "select   
  project.name,
  project.id, 
  project.notification_topic_id, 
  topic.name notification_topic_name,
  project.namespace, 
  project.description
from   
  oci_devops_project project
  left join
  oci_ons_notification_topic topic
  on project.notification_topic_id = topic.topic_id where project.id = 'ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caafqen7vittabcfzqm6ayyqwedisa2ymvblfdva2rgznpq' "
```
+--------------------+------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+-------------------------+--------------+----------------------------------------------+
| name               | id                                                                                       | notification_topic_id                                                               | notification_topic_name | namespace    | description                                  |
+--------------------+------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+-------------------------+--------------+----------------------------------------------+
| devopsproject_zApq | ocid1.devopsproject.oc1.iad.amaaaaaa6sde7caafqen7vittabcfzqm6ayyqwedisa2ymvblfdva2rgznpq | ocid1.onstopic.oc1.iad.aaaaaaaag4rsikhsiewgp63fcimxgabmr6iuk2nhkrkaks6bujw6gd2wj64a | zApq_devopstopic        | idtwlqf2hanz | DevOps Project for Instance Group deployment |
+--------------------+------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------+-------------------------+--------------+----------------------------------------------+```
</details>

### CODE REVIEW

File: oci/service.go [hunk 1]
```diff
@@ -213,6 +215,46 @@ func autoScalingService(ctx context.Context, d *plugin.QueryData, region string)
...
+	if cachedData, ok := d.ConnectionManager.Cache.Get(serviceCacheKey); ok {
+		return cachedData.(*session), nil
+	}
+
+	// get oci config info
+	ociConfig := GetConfig(d.Connection)
+
+	provider, err := getProvider(ctx, d.ConnectionManager, region, ociConfig)
+	if err != nil {
+		logger.Error("autoScalingService", "getProvider.Error", err)
```
Comment (karanpopat) [inline oci/service.go:233]:
```suggestion
		logger.Error("autoScalingService", "getProvider.Error", err)
```

File: oci/table_oci_devops_project.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+//// TABLE DEFINITION
+
+func tableDevopsProject(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:             "oci_devops_project",
+		Description:      "OCI Devops Project",
+		DefaultTransform: transform.FromCamel(),
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.SingleColumn("id"),
+			Hydrate:    getProject,
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:27]:
```suggestion
			Hydrate:    getDevopsProject,
```

File: oci/table_oci_devops_project.go [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listProjects,
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:30]:
```suggestion
			Hydrate: listDevopsProjects,
```

File: oci/table_oci_devops_project.go [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+	"strings"
+
+	oci_common "github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/devops"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/turbot/go-kit/types"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:14]: Can you please upgrade to `steampipe-plugin-sdk/v5` by rebasing with the `main`

File: oci/table_oci_devops_project.go [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+			{
+				Name:        "lifecycle_details",
+				Description: "A message describing the current state in more detail. For example, can be used to provide actionable information for a resource in Failed state.",
+				Type:        proto.ColumnType_STRING,
+			},
+
+			{
+				Name:        "lifecycle_state",
+				Description: "The current state of the project.",
+				Type:        proto.ColumnType_STRING,
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:85]:
```suggestion
				Type:        proto.ColumnType_JSON,
```

File: oci/table_oci_devops_project.go [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+//// LIST FUNCTION
+
+func listProjects(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+
+	logger.Debug("listProjects", "Compartment", compartment, "OCI_REGION", region)
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:149]:
```suggestion
	logger.Debug("listDevopsProjects", "Compartment", compartment, "OCI_REGION", region)
```

File: oci/table_oci_devops_project.go [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+			}
+		}
+		if response.OpcNextPage != nil {
+			request.Page = response.OpcNextPage
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, err
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:206]:
```suggestion
	return nil, nil
```

File: oci/table_oci_devops_project.go [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+			request.Limit = types.Int(int(*limit))
+		}
+	}
+
+	pagesLeft := true
+	for pagesLeft {
+		response, err := session.DevopsClient.ListProjects(ctx, request)
+		// see https://pkg.go.dev/github.com/oracle/oci-go-sdk/v65@v65.28.0/devops#ProjectSummary
+		if err != nil {
+			logger.Error("oci_devops_project.listProjects", "api_error", err)
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:187]:
```suggestion
			logger.Error("oci_devops_project.listDevopsProjects", "api_error", err)
```

File: oci/table_oci_devops_project.go [hunk 9]
```diff
@@ -0,0 +1,314 @@
...
+			pagesLeft = false
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getProject(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:211]:
```suggestion
func getDevopsProject(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: oci/table_oci_devops_project.go [hunk 10]
```diff
@@ -0,0 +1,314 @@
...
+			{
+				Name:        "namespace",
+				Description: "Namespace associated with the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "notification_topic_id",
+				Description: "The topic ID for the topic where project notifications will be published to.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(transformNotificationTopic),
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:96]: We can easily transform from json objects using the `transform.FromField` function

```suggestion
				Transform:   transform.FromField('NotificationConfig.TopicId'),
```

File: oci/table_oci_devops_project.go [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+	var id string
+	// if h.Item != nil {
+	// 	id = *h.Item.(queue.QueueSummary).Id
+	// } else {
+	// 	id = d.KeyColumnQuals["id"].GetStringValue()
+	// 	// Restrict the api call to only root compartment/ per region
+	// 	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+	// 		return nil, nil
+	// 	}
+	// }
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:227]:
```suggestion
```

File: oci/table_oci_devops_project.go [hunk 12]
```diff
@@ -0,0 +1,314 @@
...
+}
+
+//// HYDRATE FUNCTIONS
+
+func getProject(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getQueue")
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("getQueue", "Compartment", compartment, "OCI_REGION", region)
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:216]:
```suggestion
	logger.Debug("getDevopsProject", "Compartment", compartment, "OCI_REGION", region)
```

File: oci/table_oci_devops_project.go [hunk 13]
```diff
@@ -0,0 +1,314 @@
...
+	// 	}
+	// }
+
+	if id == "" {
+		return nil, nil
+	}
+	// Create Session
+	session, err := devopsService(ctx, d, region)
+	if err != nil {
+		logger.Error("oci_devops_project.getProject", "connection_error", err)
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:235]:
```suggestion
		logger.Error("oci_devops_project. getDevopsProject", "connection_error", err)
```

File: oci/table_oci_devops_project.go [hunk 14]
```diff
@@ -0,0 +1,314 @@
...
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "name",
+					Require: plugin.Optional,
+				},
+				{
+					Name:    "id",
+					Require: plugin.Optional,
+				},
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:43]: Would it be possible to add `lifecycle_state` as an optional `KeyColumn`

Refer - https://pkg.go.dev/github.com/oracle/oci-go-sdk/v65@v65.28.0/devops#ListProjectsRequest

File: oci/table_oci_devops_project.go [hunk 15]
```diff
@@ -0,0 +1,314 @@
...
+
+	request := devops.GetProjectRequest{
+		ProjectId: types.String(id),
+		RequestMetadata: oci_common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(d.Connection),
+		},
+	}
+	response, err := session.DevopsClient.GetProject(ctx, request)
+	if err != nil {
+		logger.Error("oci_devops_project.getQueue", "api_error", err)
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:247]:
```suggestion
		logger.Error("oci_devops_project.getDevopsProject", "api_error", err)
```

File: oci/table_oci_devops_project.go [hunk 16]
```diff
@@ -0,0 +1,314 @@
...
+	return response.Project, nil
+}
+
+// Build additional filters
+func buildProjectFilters(equalQuals plugin.KeyColumnEqualsQualMap, logger hclog.Logger) (devops.ListProjectsRequest, bool) {
+	request := devops.ListProjectsRequest{}
+	isValid := true
+
+	if equalQuals["name"] != nil && strings.Trim(equalQuals["name"].GetStringValue(), " ") != "" {
+		request.Name = types.String(equalQuals["name"].GetStringValue())
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:260]: Can you also set the other optional `KeyColumn` quals to the request

File: oci/table_oci_devops_project.go [hunk 17]
```diff
@@ -0,0 +1,314 @@
...
+	case devops.Project:
+		topicId = *d.HydrateItem.(devops.Project).NotificationConfig.TopicId
+	}
+	return topicId, nil
+}
+
+// Priority order for tags
+// 1. Defined Tags
+// 2. Free-form tags
+func projectTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:279]:
```suggestion
func devopsProjectTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
```

File: oci/table_oci_devops_project.go [hunk 18]
```diff
@@ -0,0 +1,314 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The OCID of the project.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the queue was created.",
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:65]:
```suggestion
				Description: "The date and time the project was created.",
```

File: oci/table_oci_devops_project.go [hunk 19]
```diff
@@ -0,0 +1,314 @@
...
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "time_updated",
+				Description: "The date and time the queue was last modified.",
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:71]:
```suggestion
				Description: "The date and time the project was last modified.",
```

File: oci/table_oci_devops_project.go [hunk 20]
```diff
@@ -0,0 +1,314 @@
...
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getProject(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getQueue")
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:212]:
```suggestion
	plugin.Logger(ctx).Trace("getProject")
```

File: docs/tables/oci_devops_project.md [hunk 21]
```diff
@@ -0,0 +1,66 @@
+# Table: oci_devops_project
+
+
+See: https://registry.terraform.io/providers/oracle/oci/latest/docs/data-sources/devops_project
```
Comment (karanpopat) [inline docs/tables/oci_devops_project.md:4]:
```suggestion
A DevOps project is created to successfully build and deploy applications by using the DevOps service. A project logically groups the DevOps resources needed to implement a CI/CD workflow.
```

File: docs/tables/oci_devops_project.md [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+
+
+select   
+  project.name,
+  project.id, 
+  project.notification_topic_id, 
+  project.namespace, 
+  project.description
+from   
+  oci_devops_project project
```
Comment (karanpopat) [inline docs/tables/oci_devops_project.md:66]:
```suggestion
```

Comment (karanpopat) [inline docs/tables/oci_devops_project.md:66]:
```suggestion
```

File: oci-test/tests/oci_devops_project/variables.tf [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+  #Required
+  name = var.resource_name
+  compartment_id = var.tenancy_ocid
+  notification_config {
+      #Required
+      topic_id = oci_ons_notification_topic.named_test_ons_topic_resource.id
+  }
+
+  #Optional
+
```
Comment (karanpopat) [inline oci-test/tests/oci_devops_project/variables.tf:54]:
```suggestion
```

File: oci/table_oci_devops_project.go [hunk 24]
```diff
@@ -0,0 +1,314 @@
...
+				Description: ColumnDescriptionDefinedTags,
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "freeform_tags",
+				Description: ColumnDescriptionFreefromTags,
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// // Standard Steampipe columns
```
Comment (karanpopat) [inline oci/table_oci_devops_project.go:110]:
```suggestion
			// Standard Steampipe columns
```

**Comment by karanpopat** [comment]
@lucasjellema thanks for adding this new table.

I have left a few review comments. Please let me know if you have questions.

File: oci-test/tests/oci_devops_project/test-turbot-query.sql [hunk 25]
```diff
@@ -0,0 +1,3 @@
+select title, tenant_id
+from oci.oci_apigateway_api
```
Comment (karanpopat) [inline oci-test/tests/oci_devops_project/test-turbot-query.sql:2]:
```suggestion
from oci.oci_devops_project
```

File: oci-test/tests/oci_devops_project/variables.tf [hunk 26]
```diff
@@ -0,0 +1,75 @@
+variable "resource_name" {
+  type        = string
+  default     = "steampipetest20200125"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "tenancy_ocid" {
+  type        = string
+  default     = "ocid1.tenancy.oc1..aaaaaaaag7c7slwmlvsodyym662ixlsonnihko2igwpjwwe2egmlf3gg6okq"
```
Comment (karanpopat) [inline oci-test/tests/oci_devops_project/variables.tf:9]:
```suggestion
  default     = ""
```

**Comment by lucasjellema** [comment]
> > I have a problem with the way the tests are implemented: JSON documents are stringified and then compared. For some reason, it happens that the logically same JSON documents are stringified differently and when compared are judged different making the test fail - incorrectly. The test should probably order the JSON document by key before doing the stringify (or so ChatGPT says)
> 
> @lucasjellema The JSON document is typically ordered by key by the steampipe CLI itself. Does the test fail due to the order of the keys in output?
> 
> Can you try
> 
> ```
> .output json
> select title, tenant_id from oci_devops_project;
> ```
> 
> This should return the JSON document for the output ordered by key. Please let us know if it doesn't or you face any other issues.

I am not sure I understand "output ordered by key". Can you clarify? Alphabetical order of column_name?

**Comment by misraved** [comment]
@lucasjellema apologies for the radio silence on this PR!!

The changes look good, however, this PR needs some updates, in terms of updating SDK from v4 to v5 (this should fix the lint failures as well).

Would like to get this PR merged and pushed to the hub 🎉 .

**Comment by ParthaI** [comment]
Addressed in https://github.com/turbot/steampipe-plugin-oci/pull/523

**Pull request closed without merging**